packages/format-library/src/text-color/inline.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createRef, useMemo } from '@wordpress/element';
+import { select } from '@wordpress/data';
+import {
+	withSpokenMessages,
+	ColorPalette,
+} from '@wordpress/components';
+import { getRectangleFromRange } from '@wordpress/dom';
+import {
+	applyFormat,
+	removeFormat,
+	getActiveFormat,
+} from '@wordpress/rich-text';
+import { URLPopover } from '@wordpress/block-editor';
+
+const URLPopoverAtLink = ( { isActive, addingLink, value, ...props } ) => {
```

Comment:
I'm starting to see this pattern across several formats, can we factorize it somehow?

Reply:
@youknowriad Sorry for late response. It's my first contribution and not quite familiar on how creating components work on core. Do I need to create separate PR?

Reply:
Yeah, I think it can be a separate PR. @ellatrix might have thoughts as well as she used this in other formats.

Reply:
@youknowriad For now, I'll just change `URLPopoverAtLink` to different name so make it less confusing.